test/specs/components/Button/Button-test.tsx

```diff
@@ -50,6 +52,51 @@ describe('Button', () => {
       })
     })
 
+    describe('AXE validation', () => {
+      describe('icon button must have textual representation for screen readers', () => {
+        test('with title', async () => {
+          const html = ReactDOMServer.renderToString(<Button icon="books" title="testing button" />)
+          const results = await axe(html)
+          expectAxe(results).toHaveNoViolations()
+        })
+
+        test('with aria-label attribute', async () => {
+          const html = ReactDOMServer.renderToString(
+            <Button icon="books" aria-label="testing button" />,
+          )
+          const results = await axe(html)
+          expectAxe(results).toHaveNoViolations()
+        })
+
+        test('with aria-labelledby attribute', async () => {
+          const html = ReactDOMServer.renderToString(
+            <div>
+              <Button icon="books" aria-labelledby="tstBtn" />
+              <span id="tstBtn" aria-label="testing button" />
+            </div>,
+          )
+          const results = await axe(html)
+          expectAxe(results).toHaveNoViolations()
+        })
+      })
+
+      describe('different buttons variants', () => {
+        test('button', async () => {
+          const html = ReactDOMServer.renderToString(<Button>Simple test button</Button>)
```

Comment:
we are using the same pattern over an over again - essentially
- rendering some portion of JSX to HTML string
- requesting aXe check
- validating result

I would rather suggest to make it all being a function that will be exported by 'expectAxe' module (we might consider to rename it to 'checkAxe' in this case)